### PR TITLE
Add own reason to $Shape

### PR DESCRIPTION
**Pull request by goodmind**
As in the title

### CODE REVIEW

File: src/typing/flow_js.ml [hunk 1]
```diff
@@ -4444,17 +4444,17 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
...
      * match an object type with a subset of the props in the type being
      * destructured. It would be complicated and confusing to use a function for
      * this.
      *
      * This invariant is important for the React setState() type definition. *)
-    | _, UseT (use_op, ShapeT o) ->
+    | _, UseT (use_op, ShapeT (reason_op, o)) ->
       add_output cx ~trace
-        (Error_message.EIncompatibleWithShape (reason_of_t l, reason_of_t o, use_op))
+        (Error_message.EIncompatibleWithShape (reason_op, reason_of_t o, use_op))
```
Comment (mroch) [inline src/typing/flow_js.ml]: i think this should probably be `(reason_of_t l, reason_op, use_op)`

Comment (goodmind) [reply src/typing/flow_js.ml]: done

**Comment by goodmind** [comment]
```
is incompatible with `$Shape` of shape of `Foo`
```
well

**Comment by goodmind** [comment]
Probably I can just remove `EIncompatibleWithShape` entirely

**Pull request closed without merging**